cayenne/src/main/java/org/apache/cayenne/access/translator/select/OrderingStage.java

```diff
@@ -71,4 +85,26 @@ private void processOrdering(QualifierTranslator qualifierTranslator, Translator
         context.getSelectBuilder().orderBy(orderingNodeBuilder);
     }
 
+    private Optional<Node> getResultColumn(TranslatorContext context, OrderingNodeBuilder orderBuilder) {
+        String orderStr = getSqlString(orderBuilder);
+
+        return context.getResultNodeList().stream()
+                .map(result -> result.getNode())
+                // the column may have an alias so just check that the orderStr part matches
+                .filter(resultNode -> getSqlString(node(resultNode)).startsWith(orderStr))
+                .findFirst();
```

Comment:
This seems a bit heavy, but we could worry about performance later.

Reply:
I thought it's a bit heavy as well. Will need guidance later .....